### ISSUE TITLE
Cmd+K Popup

### ISSUE THREAD

**Issue by vmatsiiako** [Infisical/infisical#675]
### Feature description
When people click cmd+K (or ctrl+K on Windows), they should see such a modal popup:

<img width="1471" alt="Screenshot 2023-06-19 at 13 11 32" src="https://github.com/Infisical/infisical/assets/78047717/af20a784-3b44-40ec-a9d1-1230ff8ee9af">

Here are some ideas for what actions can be included in this modal: 
- [ ] Create a new project
- [ ] Add a folder
- [ ] Add a secret
- [ ] Add a new member to a project
- [ ] Invite a user to Infisical
- [ ] etc

In future, if a person is exploring a particular environment, we should have option to 
- [ ] Replicate/Inherit an environment

### Why would it be useful?
It provides an amazing developer experience, and speeds up many actions.

**Comment by vmatsiiako** [comment]
@DJaegerScript yes, it is! Please join our Slack if you need any help: https://infisical.com/slack

**Comment by MrPoliteGrizlyM** [comment]
Hey @vmatsiiako is this feature still open? lmk if anyone else is working on this one.

**Comment by vmatsiiako** [comment]
Hey @MrPoliteGrizlyM! I don't think anyone is working on it at the moment! Would be awesome if you could help!

**Comment by MrPoliteGrizlyM** [comment]
Alright, sounds good! I will get this done by Friday 🚀
I will reply to this thread if anything changes.
Can you elaborate more on this:
> In future, if a person is exploring a particular environment, we should have option to
> Replicate/Inherit an environment

Examples would be very helpful, thanks!

**Comment by vmatsiiako** [comment]
An example of that is: imagine you are currently in the explore for a particular environment in a certain project. You could click cmd+K and select "duplicate environment" under a different name

**Comment by MrPoliteGrizlyM** [comment]
Gotcha, just want to confirm:
Do you mean by "environment" the dev/test/staging/production/some_custom_env? Lets say you have couple of folders and secrets in dev env. By clicking "Duplicate environment" in dev, all of the content will be copied to the selected env or new created env (modal diolog will be shown to which exactly env this should be copied).

**Comment by vmatsiiako** [comment]
Yes, indeed!

**Comment by MrPoliteGrizlyM** [comment]
Hey @vmatsiiako, thank you for your confirmation about the "duplicate environment" feature, I'm on it! I will implement this right after I finish the basic actions like "Create Secret", "Create Project" etc.
Implementing this feature takes a bit longer than I expected. Currently I'm exploring how should I compose the action modals, so new action modals can be added much easier. I will let you know the status on Monday! 
For now, I recorded a short video on the CMD+K feature UI. Could you please review it and give me feedback? Let me know if any styles/icons needs to changed. Thank you!

https://github.com/Infisical/infisical/assets/37931126/d1e8d2d3-d380-4271-8a7d-567e846fafef

I'm planning to make the exact same UX just like in Linear: Click Action -> Close Window -> Show corresponding action modal (CMD+K window can't be open until this modal is closed). I'm open to discuss this!

**Comment by vmatsiiako** [comment]
@MrPoliteGrizlyM i just learn from @dangtony98 and @maidul98 that they are working on the revamp to the dashboard which might have a lot of merge conflicts with this issue. @dangtony98 what do you think we should do?

**Comment by MrPoliteGrizlyM** [comment]
@vmatsiiako I see, in this case I will wait until you guys finish the dashboard. Thanks for letting me know!
I haven't pushed anything big yet, I've pushed only the [CMDK demo](https://github.com/MrPoliteGrizlyM/infisical/tree/feature/cmd+k). Only the frontend/src/pages/_app.tsx is modified. 
 @dangtony98 @maidul98 Are you guys planning to use all of the modals as components? I have found that couple of modals are hardcoded in the pages for example "Create a new project" on the org/[id]/overview page. Some of them are in the separate file for example the "Create secret" modal in the "CreateSecreForm" component. It will be very useful if we could reuse all of the modals for the CMDK feature. I was thinking of implementing the "ModalsFactory" myself that will contain all of the modals in the project. Let me know what you think 🙂
